connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/HttpCommonResultResponseHandler.java

```diff
@@ -88,7 +88,30 @@ public HttpCommonResult handleResponse(ClassicHttpResponse response) {
     return new HttpCommonResult(code, headers, null, reason);
   }
 
-  private Document handleFileResponse(Map<String, String> headers, byte[] content) {
+  private Map<String, Object> formatHeaders(Header[] headersArray) {
```

Comment:
Could be a static utility method that receives the `isHeaderGroupingEnabled` as a parameter.